packages/nextjs/src/common/wrapApiHandlerWithSentry.ts

```diff
@@ -81,8 +81,10 @@ export function wrapApiHandlerWithSentry(apiHandler: NextApiHandler, parameteriz
                   // eslint-disable-next-line @typescript-eslint/unbound-method
                   res.end = new Proxy(res.end, {
                     apply(target, thisArg, argArray) {
-                      setHttpStatus(span, res.statusCode);
-                      span.end();
+                      if (span.isRecording()) {
```

Comment:
Also noticed this in tests, we sometimes seemed to double-end spans, so guarding against this here.